distributed/tests/test_client.py

```diff
@@ -3365,16 +3365,20 @@ def test_default_get(loop_in_thread):
         assert dask.base.get_scheduler() == pre_get
 
 
-@gen_cluster(client=True)
-async def test_ensure_default_client(c, s, a, b):
-    assert c is default_client()
-
-    async with Client(s.address, set_as_default=False, asynchronous=True) as c2:
+@gen_cluster()
+async def test_ensure_default_client(s, a, b):
+    c = await Client(s.address, asynchronous=True)
+    try:
```

Comment:
Turns out we can only set current when entering ctx when `set_as_default` is also set. Otherwise this breaks tests like this.

That also makes sense considering that `default_client` and `Client.current` are effectively aliases